Mathlib/Data/Set/Monotone.lean

```diff
@@ -0,0 +1,177 @@
+/-
+Copyright (c) 2014 Jeremy Avigad. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Avigad, Andrew Zipperer, Haitao Zhang, Minchao Wu, Yury Kudryashov
+-/
+import Mathlib.Data.Set.Function
+import Mathlib.Data.Set.Prod
+import Mathlib.Logic.Function.Conjugate
+
+/-!
+# Monotone functions over sets
+-/
+
+variable {α β γ : Type*} {ι : Sort*} {π : α → Type*}
```

Comment:
I don't see ι or π being used below; please check what you can trim from the `variable`s and `open`s

Reply:
Ooh good point. I wonder if #17261 would have detected this for me!

Reply:
I am quite confident that it would have, but the linter is still work-in-progress.

Reply:
I decided to run the linter on the files modified in this PR.  Besides lots of accumulated unused variables upstream of the files here, this is the report:
```
ℹ [333/334] Replayed Mathlib.Data.Set.Function
info: ././././Mathlib/Data/Set/Function.lean:169:28: 't' is unused
info: ././././Mathlib/Data/Set/Function.lean:170:45: 'b' is unused
info: ././././Mathlib/Data/Set/Function.lean:453:10: 'U' is unused
info: ././././Mathlib/Data/Set/Function.lean:169:58: 'f' is unused
info: ././././Mathlib/Data/Set/Function.lean:169:46: 'p' is unused
info: ././././Mathlib/Data/Set/Function.lean:170:3: 'f'' is unused
info: ././././Mathlib/Data/Set/Function.lean:169:81: 'g₁' is unused
info: ././././Mathlib/Data/Set/Function.lean:169:33: 't₂' is unused
info: ././././Mathlib/Data/Set/Function.lean:169:84: 'g₂' is unused
info: ././././Mathlib/Data/Set/Function.lean:169:30: 't₁' is unused
info: ././././Mathlib/Data/Set/Function.lean:236:66: 'f₃' is unused
info: ././././Mathlib/Data/Set/Function.lean:170:10: 'f₂'' is unused
info: ././././Mathlib/Data/Set/Function.lean:170:24: 'g'' is unused
info: ././././Mathlib/Data/Set/Function.lean:170:6: 'f₁'' is unused
ℹ [334/334] Replayed Mathlib.Data.Set.Monotone
info: ././././Mathlib/Data/Set/Monotone.lean:60:12: 's₁' is unused
info: ././././Mathlib/Data/Set/Monotone.lean:14:26: 'ι' is unused
info: ././././Mathlib/Data/Set/Monotone.lean:14:38: 'π' is unused
info: ././././Mathlib/Data/Set/Monotone.lean:60:33: 'f₂' is unused
info: ././././Mathlib/Data/Set/Monotone.lean:159:60: 's' is unused
info: ././././Mathlib/Data/Set/Monotone.lean:159:62: 't' is unused
info: ././././Mathlib/Data/Set/Monotone.lean:159:36: 'f' is unused
info: ././././Mathlib/Data/Set/Monotone.lean:60:30: 'f₁' is unused
info: ././././Mathlib/Data/Set/Monotone.lean:159:10: 'fa' is unused
info: ././././Mathlib/Data/Set/Monotone.lean:159:23: 'fb' is unused
info: ././././Mathlib/Data/Set/Monotone.lean:159:48: 'g' is unused
```

Reply:
Thanks, super helpful!

Reply:
If you want, I can also push the changes to the other file: I cannot add them as suggestions, since they are on unmodified lines.

Reply:
Yes, please feel free!

Reply:
Done!